lib/auth/auth.go

```diff
@@ -4378,6 +4378,47 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 	return req, nil
 }
 
+// getAccessRequestPromotions will return potential access list promotions for an access request.
+func (a *Server) getAccessRequestPromotions(ctx context.Context, req types.AccessRequest) (types.AccessRequest, *types.AccessRequestAllowedPromotions) {
```

Comment:
I'd rename the function. Get suggests that something is being fetched. Here we're generating/creating a new resource.